docs/source/public_api/problem_zoo.rst

```diff
@@ -0,0 +1,24 @@
+probnum.problem_zoo
+========================
+
+Problem zoo for probabilistic numerical methods.
+
+Subpackage offering implementations of standard example problems or convenient
+interfaces to benchmark problems for probabilistic numerical methods. These test
+problems are meant for rapid experimentation and prototyping.
+
+.. automodapi:: probnum.problem_zoo
+    :no-heading:
+    :no-main-docstr:
+
+probnum.problem_zoo.ode
```

Comment:
sorry, it seems that I missed this last time. Do you think that until there is significant overlap between `quad`, `linalg`, etc., it would be best to shadow the module structure in here? In this case this would probably be better called `diffeq`. Though, we can also postpone this until very soon when I move the ivp examples in here.

Reply:
I agree that we should mostly shadow the package structure here.

Let's discuss this when talking about the API today, but i think it might make sense to separate ODEs and PDEs since the methods are so different. So maybe in the main package we also want `probnum.ode(s)` and `probnum.pde(s)`. 

I or Marvin may add discretized PDE problems to the model zoo in the near future (think Galerkin methods) which I think intended to put in `probnum.model_zoo.pde`.